Make sleep timer use a GLib timeout to avoid issues with sleeping after the end of a chapter

stevenleadbeater opened the pull request:
* Added tar ball for convenience testing on phones
* Fixed build errors with peewee
* Use GLib timer to count past chapter ends
* make methods that don't use locals static to avoid warnings

rdbende commented:
Thanks! I reverted some of your changes, because I had fixed them on `master` already. The rest looks good to me, but I just realized, that with this solution, the countdown will continue even when the playback is paused. Also, I can't reproduce #646 anymore, maybe I did fix it somehow in #973?

stevenleadbeater commented:
> Thanks! I reverted some of your changes, because I had fixed them on `master` already. The rest looks good to me, but I just realized, that with this solution, the countdown will continue even when the playback is paused. Also, I can't reproduce #646 anymore, maybe I did fix it somehow in #973?

Actually having the timer continue when the book is paused is kind of normal for an audiobook player, if I can't sleep and go down stairs to make a drink or something then come back and unpause, other apps usually carry the sleep timer on... can't say for definite which but it kind of feels natural to me? this is kind of airy fairy stuff though, I can see why cancelling the timer might be preferred too?
